### ISSUE TITLE
Uploading image with metadata

### ISSUE THREAD

**Issue by GoogleCodeExporter** [barsu8184/google-gdata#103]
```
In ASP.NET we are able to only upload image with no metadata and developer 
has to send 2 requests (Insert and Update) to completely update picture in 
album. So we need a way to upload picture and metadata with a single 
request. 
It's already done in Java client library so it would be nice if you do 
something like

PhotoEntry entry = new PhotoEntry();
entry.SetDescription("This is sample description");
entry.SetAuthor("John Doe");
entry.SetDate(DateTime.Now());
entry.Media.SetSource("c:\\My Photos\\P1111.JPG");
picasaService.Insert(entry);

Thanks.
```

Original issue reported on code.google.com by `dpro...@gmail.com` on 29 Dec 2007 at 8:45

**Comment by GoogleCodeExporter** [comment]
```
This is a pure optimisation, and requires MIME code, which does not really 
exist inside .NET (compared to java). Is 
there any requirement on your side, or is that a pure convienience thing? 
```

Original comment by `fman...@gmail.com` on 9 Jan 2008 at 7:03
- Changed state: **Reviewed**

**Comment by GoogleCodeExporter** [comment]
```
Thanks for reply.
In my project everything I need is upload picture and description and sometimes 
Geodata. This is why I think 2 requests are still too much for such simple 
upload 
operation. 
I think that also would be more convenient for everyone who uses ASP.NET. As a 
possible way to meet those MIME requirements would be if we do some action with 
MIME 
manually and send result as 2nd parameter to SetSource method. 
```

Original comment by `dpro...@gmail.com` on 9 Jan 2008 at 3:21

**Comment by GoogleCodeExporter** [comment]
```
That's the problem. Creating a Mime Multipart envelope is not trivial. I don't 
want to put Mime code inside 
the .NET library per se, as the likelyhood that there is desire to use it out 
of context is too great, and i don't 
have the resources to support a fully fledge MIME project. I find it personally 
unsatisfying that there is no 
MIME code in .NET nor that there is a good opensource project that i could just 
reuse for this. 

Yes, 2 requests is worse than one. And given time, the code will probably be 
written, but it is not high on 
my priority list right now, considering the other things on the horizon. 

So i am in agreement with you, the priorities are just not a perfect match. 
```

Original comment by `fman...@gmail.com` on 11 Jan 2008 at 1:20

**Comment by GoogleCodeExporter** [comment]
```
Thank you for reply. 
I understand your point. And I hope that the day when it's done will come 
sooner 
than we expect :) 
Good luck and thank you for your work.
```

Original comment by `dpro...@gmail.com` on 11 Jan 2008 at 5:00

**Comment by GoogleCodeExporter** [comment]
```
Hi fmantek, may I know what do you mean by MIME code here? I happened to make an
ASP.NET class to upload image with metadata on Sep 2007. But recently the class 
does
not working anymore, I am wondering if PWA group has changed their web service.

Please check the question that I posted in the group:
http://groups.google.com/group/Google-Picasa-Data-API/browse_thread/thread/de868
65b3ea47e63

Thanks.
```

Original comment by `atcup...@gmail.com` on 17 Mar 2008 at 2:02

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
FYI, my issue is resolved. For further details, you could refer to the URL I 
posted in
previous comment, thanks.
```

Original comment by `atcup...@gmail.com` on 29 Mar 2008 at 4:40

**Comment by GoogleCodeExporter** [comment]
```
This is fixed with the 1.2.0.0 update
```

Original comment by `fman...@gmail.com` on 17 Jun 2008 at 3:52
- Changed state: **Fixed**
